e2e/etcd_release_migrate_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2016 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/coreos/etcd/pkg/testutil"
+)
+
+// TestReleaseMigrate ensures that changes to master branch does not affect the
+// migration from latest etcd releases.
+func TestReleaseMigrate(t *testing.T) {
+	releaseBinaryPath := os.Getenv("RELEASE_BINARY")
```

Comment:
the prior release could just be extracted to ../bin/etcd-last-release instead of having to pass around an environment variable; if the binary doesn't exist, skip